// web3-errors/dist/errors/generic_errors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AbiError = exports.OperationAbortError = exports.OperationTimeoutError = exports.MethodNotImplementedError = exports.FormatterError = exports.InvalidNumberOfParamsError = void 0;
const error_codes_1 = require("../error_codes");
const web3_error_base_1 = require("../web3_error_base");
class InvalidNumberOfParamsError extends web3_error_base_1.Web3Error {
    constructor(got, expected, method) {
        super(`Invalid number of parameters for "${method}". Got "${got}" expected "${expected}"!`);
        this.got = got;
        this.expected = expected;
        this.method = method;
        this.code = error_codes_1.ERR_PARAM;
    }
    toJSON() {
        return Object.assign(Object.assign({}, super.toJSON()), { got: this.got, expected: this.expected, method: this.method });
    }
}
exports.InvalidNumberOfParamsError = InvalidNumberOfParamsError;
class FormatterError extends web3_error_base_1.Web3Error {
    constructor() {
        super(...arguments);
        this.code = error_codes_1.ERR_FORMATTERS;
    }
}
exports.FormatterError = FormatterError;
class MethodNotImplementedError extends web3_error_base_1.Web3Error {
    constructor() {
        super("The method you're trying to call is not implemented.");
        this.code = error_codes_1.ERR_METHOD_NOT_IMPLEMENTED;
    }
}
exports.MethodNotImplementedError = MethodNotImplementedError;
class OperationTimeoutError extends web3_error_base_1.Web3Error {
    constructor() {
        super(...arguments);
        this.code = error_codes_1.ERR_OPERATION_TIMEOUT;
    }
}
exports.OperationTimeoutError = OperationTimeoutError;
class OperationAbortError extends web3_error_base_1.Web3Error {
    constructor() {
        super(...arguments);
        this.code = error_codes_1.ERR_OPERATION_ABORT;
    }
}
exports.OperationAbortError = OperationAbortError;
class AbiError extends web3_error_base_1.Web3Error {
    constructor() {
        super(...arguments);
        this.code = error_codes_1.ERR_ABI_ENCODING;
    }
}
exports.AbiError = AbiError;
